### ISSUE TITLE
1.16.3 Server crash on chunk generation.

### ISSUE THREAD

**Issue by Voknehzyr** [paulevsGitch/BetterNether#169]
I think the error is enough to get some idea of whats happening wrong.

```[21:35:26] [Server thread/ERROR]: Failed to save chunk [0, 32]
java.util.concurrent.CompletionException: net.minecraft.class_148: Exception generating new chunk
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273) ~[?:1.8.0_252]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280) ~[?:1.8.0_252]
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:975) ~[?:1.8.0_252]
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940) ~[?:1.8.0_252]
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456) ~[?:1.8.0_252]
        at net.minecraft.class_3900.method_17634(class_3900.java:58) ~[intermediary-server.jar:?]
        at net.minecraft.class_3846.method_16907(class_3846.java:94) ~[intermediary-server.jar:?]
        at net.minecraft.class_3846.method_16900(class_3846.java:137) ~[intermediary-server.jar:?]
        at net.minecraft.class_3846.run(class_3846.java:105) ~[intermediary-server.jar:?]
        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[?:1.8.0_252]
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:1.8.0_252]
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[?:1.8.0_252]
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[?:1.8.0_252]
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[?:1.8.0_252]
Caused by: net.minecraft.class_148: Exception generating new chunk
        at net.minecraft.class_3898.method_17225(class_3898.java:579) ~[intermediary-server.jar:?]
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[intermediary-server.jar:?]
        at net.minecraft.class_3898.method_17224(class_3898.java:562) ~[intermediary-server.jar:?]
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966) ~[?:1.8.0_252]
        ... 11 more
Caused by: java.lang.ClassCastException: net.minecraft.class_3233 cannot be cast to net.minecraft.class_1937
        at net.Shad0w.Allure.Blocks.Speleothem.SpeleothemBlock.method_9558(SpeleothemBlock.java:49) ~[Allure-1.2.7.jar:?]
        at net.minecraft.class_4970$class_4971.method_26184(class_4970.java:926) ~[intermediary-server.jar:?]
        at paulevs.betternether.world.BNWorldGenerator.fixBlocks(BNWorldGenerator.java:482) ~[better-nether-4.0.1.jar:?]
        at paulevs.betternether.world.BNWorldGenerator.cleaningPass(BNWorldGenerator.java:427) ~[better-nether-4.0.1.jar:?]
        at net.minecraft.class_2794.handler$zil000$customPopulate(class_2794.java:1064) ~[intermediary-server.jar:?]
        at net.minecraft.class_2794.method_12102(class_2794.java) ~[intermediary-server.jar:?]
        at net.minecraft.class_2806.method_12151(class_2806.java:89) ~[intermediary-server.jar:?]
        at net.minecraft.class_2806.method_12154(class_2806.java:219) ~[intermediary-server.jar:?]
        at net.minecraft.class_3898.method_17225(class_3898.java:568) ~[intermediary-server.jar:?]
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[intermediary-server.jar:?]
        at net.minecraft.class_3898.method_17224(class_3898.java:562) ~[intermediary-server.jar:?]
        at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966) ~[?:1.8.0_252]
        ... 11 more```

**Comment by paulevsGitch** [comment]
ClassCastException - net.Shad0w.Allure.Blocks.Speleothem.SpeleothemBlock.method_9558. Looks like it is not relative to BN, but this happens in cleaning pass. You can try to disable cleaning pass or report this issue to author of conflicting mod

**Comment by Voknehzyr** [comment]
> ClassCastException - net.Shad0w.Allure.Blocks.Speleothem.SpeleothemBlock.method_9558. Looks like it is not relative to BN, but this happens in cleaning pass. You can try to disable cleaning pass or report this issue to author of conflicting mod

I disabled cleaning pass but it still throws exception including it so I have no idea how to fix that without compatibility :D

**Comment by paulevsGitch** [comment]
This is not a BN error actually. You need to write a report to another mod author

**Comment by theishiopian** [comment]
out of curiosity, did you ever figure out what the error on his end was? seems rather odd that an error in the cleaning pass can be thrown if that pass is disabled

**Comment by paulevsGitch** [comment]
The error is in blockstate, seems like Allure changes something in blockstate work. Error is in cast to BlockState (net.minecraft.class_1937) from some another class (net.minecraft.class_3233) in SpeleothemBlock.method_9558 on line 49 (SpeleothemBlock.java)

**Comment by paulevsGitch** [comment]
As this error are in one of basic methods it will crash independently from cleaning pass

**Comment by LeLobster** [comment]
> it will crash independently from cleaning pass

This is actually not the case for me with `better-nether-4.1.3` & `Allure-1.2.7`    
I've tried in a fresh profile with only Fabric, BN and Allure and if you disable `terrain_cleaning_pass` and `world_fixing_pass` then the Nether generates without crashing (albeit a bit messy)

**Comment by ProsperCraft** [comment]
https://www.curseforge.com/minecraft/mc-mods/allure/issues/8

**Comment by LeLobster** [comment]
Looks like the Allure dev returned and has this is fixed with 1.2.8 which he/she uploaded yesterday  
Haven't yet tested it myself though
